pkg/collections/maps/utils.go

```diff
@@ -50,3 +55,34 @@ func Intersect[K comparable, V any](source map[K]V, maps ...map[K]V) map[K]V {
 
 	return res
 }
+
+// MapKeys returns a slice of all keys in the map.
+// The order is unspecified.
+// The keys are not deep-copied, so changes to them could affect the original map.
+func MapKeys[K comparable, V any](m map[K]V) []K {
+	keys := make([]K, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	return keys
+}
```

Comment:
Could we instead use the `Keys` function from the standard package here as well: https://pkg.go.dev/maps#Keys? Or does this not work, with what you have planned?

Reply:
`maps.Keys`/`maps.Values` don't return a slice, but an `iter.Seq` instead. This is better when you are iterating over the returned values, but it is annoying, if you want to pass the returned values into a function that takes a slice, because you would have to iterate over them in order to build the slice.

Therefore I think that it is fine to have these `MapKeys` and `MapValues` implementations in addition to `map.Keys` and `map.Values`.

WDYT?

Reply:
The `iter.Seq` return value would work with [`slices.Collect`](https://pkg.go.dev/slices#Collect) from the standard package to turn it into a slice. Would that work too in our case?